WalletWasabi/Legal/LegalDocuments.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Helpers;
+using WalletWasabi.Logging;
+using WalletWasabi.WebClients.Wasabi;
+
+namespace WalletWasabi.Legal
+{
+	public class LegalDocuments
+	{
+		public const string EmbeddedFileName = "LegalDocuments.txt";
+		public const string LegalFolderName = "Legal";
+		public const string AssetsFoldername = "Assets";
+		public static readonly string EmbeddedFilePath = Path.Combine(EnvironmentHelpers.GetFullBaseDirectory(), LegalFolderName, AssetsFoldername, EmbeddedFileName);
+		public string FilePath { get; }
+		public Version Version { get; }
+
+		public static async Task<LegalDocuments> TryLoadAgreedAsync(string dataDir)
+		{
+			var legalFolderPath = Path.Combine(dataDir, LegalFolderName);
+			IoHelpers.EnsureDirectoryExists(legalFolderPath);
+
+			int fileCount = Directory.EnumerateFileSystemEntries(legalFolderPath).Count();
+			// If more than one file found, then something strange happened, delete the dir and start from zero.
+			if (fileCount > 1)
+			{
+				Logger.LogInfo("Multiple legal docs detected. Recovering empty legal directory...");
+				await IoHelpers.DeleteRecursivelyWithMagicDustAsync(legalFolderPath).ConfigureAwait(false);
+				IoHelpers.EnsureDirectoryExists(legalFolderPath);
+			}
+
+			await TryEnsureBackwardsCompatibilityAsync(dataDir).ConfigureAwait(false);
+
+			var existingFilePath = Directory.EnumerateFiles(legalFolderPath, "*.txt", SearchOption.TopDirectoryOnly).FirstOrDefault();
+			if (existingFilePath is { })
+			{
+				var verString = Path.GetFileNameWithoutExtension(existingFilePath);
+				if (Version.TryParse(verString, out Version version))
+				{
+					string filePath = existingFilePath;
+					return new LegalDocuments(version, filePath);
+				}
+				else
+				{
+					File.Delete(existingFilePath);
+				}
+			}
+
+			return null;
+		}
+
+		private static async Task TryEnsureBackwardsCompatibilityAsync(string dataDir)
+		{
+			var legalFolderPath = Path.Combine(dataDir, LegalFolderName);
+			IoHelpers.EnsureDirectoryExists(legalFolderPath);
+
+			int fileCount = Directory.EnumerateFileSystemEntries(legalFolderPath).Count();
+			// If no file found then to ensure backwards compatibility we can check if a wallet file already exists.
+			// If a wallet file already exist, that means the user already accepted 1.0 at one point with the old system.
```

Comment:
What if the user manually copies a wallet file there?

Reply:
We are working on the new legal docs. Now it is being reviewed by a third party.

Reply:
No need to handle such edge case. Also I would not like this to be merged without legal docs 2.0, I just implemented it just in case.